arcgis-ios-sdk-samples/Edit data/Edit features with feature-linked annotation/EditFeaturesWithFeatureLinkedAnnotationViewController.swift

```diff
@@ -0,0 +1,253 @@
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class EditFeaturesWithFeatureLinkedAnnotationViewController: UIViewController {
+    @IBOutlet weak var mapView: AGSMapView! {
+        didSet {
+            // Create the map with a light gray canvas basemap centered on Loudoun, Virginia.
+            mapView.map = AGSMap(basemapType: .lightGrayCanvasVector, latitude: 39.0204, longitude: -77.4159, levelOfDetail: 18)
+            // Set the touch delegate.
+            mapView.touchDelegate = self
+            loadGeodatabase()
+        }
+    }
+    // The feature that has been selected.
+    var selectedFeature: AGSFeature?
+    // A geodatabase to be loaded.
+    var geodatabase: AGSGeodatabase?
+    // The returned cancelable after identifying the layers.
+    var identifyLayers: AGSCancelable?
+    
+    func loadGeodatabase() {
+        // Obtain the geodatabase URL from portal data.
+        let geodatabaseURL = Bundle.main.url(forResource: "loudoun_anno", withExtension: "geodatabase")!
+        geodatabase = AGSGeodatabase(fileURL: geodatabaseURL)
+        guard let geodatabase = geodatabase else { return }
+        // Load a geodatabase from portal data.
+        geodatabase.load { [weak self] (error: Error?) in
+            guard let self = self else { return }
+            if let error = error {
+                self.presentAlert(error: error)
+            } else {
+                guard let map = self.mapView.map else { return }
+                // Create feature layers from tables in the geodatabase.
+                let featureTableNames = ["ParcelLines_1", "Loudoun_Address_Points_1"]
+                let featureTables = featureTableNames.compactMap { geodatabase.geodatabaseFeatureTable(withName: $0) }
+                let featureLayers = featureTables.map(AGSFeatureLayer.init)
+                // Add the feature layers to the map.
+                map.operationalLayers.addObjects(from: featureLayers)
+                // Create annotation layers from tables in the geodatabase.
+                let annotationTableNames = ["ParcelLinesAnno_1", "Loudoun_Address_PointsAnno_1"]
+                let annotationTables = annotationTableNames.compactMap { geodatabase.geodatabaseAnnotationTable(withTableName: $0) }
+                let annotationLayers = annotationTables.map(AGSAnnotationLayer.init)
+                // Add the annotation layers to the map.
+                map.operationalLayers.addObjects(from: annotationLayers)
+            }
+        }
+    }
+    
+    func selectFeature(at: CGPoint) {
+        // Clear any previously selected features.
+        clearSelection()
+        
+        // Identify across all layers.
+        identifyLayers = mapView.identifyLayers(atScreenPoint: at, tolerance: 10.0, returnPopupsOnly: false) { [weak self] (results: [AGSIdentifyLayerResult]?, error: Error?) in
+            guard let self = self, let results = results else { return }
```

Comment:
Guarding that `results` is non-nil here means that lines `70` through `74` are useless. Either `results` or `error` will be non-nil, but not both. If results is non-nil, the error is `nil`. Instead, the `if` statement should really be an `if…else if` statement, with one branch for when `results` is non-nil and the other branch for when `error` is non-nil.